### ISSUE TITLE
Happypack 4.0.0-beta.2 throws when used with cache-loader 1.0.3

### ISSUE THREAD

**Issue by IllusionMH** [amireh/happypack#184]
Webpack: v3.5.5
Happypack: v4.0.0-beta.2
cache-loader: v1.0.3

When we use HappyPack with `babel-loader` alone, there is no problems, but when I add `cache-loader` build fails with errors from HappyPack package.

Plugin initialization
```js
...
plugins: [
    new HappyPack({
        id: 'js',
        threadPool: happyThreadPool,
        loaders: ['cache-loader', 'babel-loader']
    }),
...
]
...
```

Error
```
ERROR in <entry point>.js
Module build failed: TypeError: fn.apply is not a function
    at applySyncOrAsync (<project directory>\node_modules\happypack\lib\applyLoaders.js:342:21)
    at apply (<project directory>\node_modules\happypack\lib\applyLoaders.js:269:5)
    at <project directory>\node_modules\happypack\lib\applyLoaders.js:132:7
    at applyPitchLoader (<project directory>\node_modules\happypack\lib\applyLoaders.js:185:14)
    at <project directory>\node_modules\happypack\lib\applyLoaders.js:219:11
    at <project directory>\node_modules\cache-loader\dist\index.js:95:7
    at FSReqWrap.cb [as oncomplete] (fs.js:258:19)
```
Looks like it its because `cache-loader` transpiles code from ES2015 export, however runs fine with it.
```js
...
rules: [
    {
        test: /\.jsx?$/,
        exclude: /(node_modules|3p)/,
        // use: 'happypack/loader?id=js'
        use: ['cache-loader', 'babel-loader']
    }
...
```
Is there a working example on how `HappyPack v4` can be used with `cache-loader`?

**Comment by amireh** [comment]
Thanks for reporting! The fix should land in the `4.0.0-beta.5` release, I'll close the issue when that happens.

**Comment by IllusionMH** [comment]
@amireh thanks for the fix!
Checked with current `master` and works fine now + noticeable positive effect on build time for our config.

Do you have any ETA for `4.0.0-beta.5` release? (will help with planning migration timeline)

**Comment by amireh** [comment]
We can do it today since this fix has been confirmed.

**Comment by IllusionMH** [comment]
@amireh that would be great if you don't have any other pending changes for `beta.5`.

**Comment by amireh** [comment]
I've just published `happypack@4.0.0-beta.5` to npm.
